internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -81,6 +81,12 @@ func resourceKubernetesClusterNodePool() *pluginsdk.Resource {
 					ForceNew:     true,
 					ValidateFunc: validation.StringIsNotEmpty,
 				},
+				"host_group_id": {
+					Type:         pluginsdk.TypeString,
+					Optional:     true,
+					ForceNew:     true,
+					ValidateFunc: azure.ValidateResourceID,
```

Comment:
Validate func refs to https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/compute/linux_virtual_machine_resource.go#L156